### PR TITLE
Minor Improvements to Collaborative Editing

### PR DESCRIPTION
**Pull request by darkruby501**
- Clicking away from the title input on post edit form causes the title to be saved
- Notifications for in-line comments now display appropriate message for author and others
- Version History button only appears for posts that have been saved (doesn't show on New Question/Post forms)

### CODE REVIEW

File: packages/lesswrong/components/editor/EditTitle.tsx [hunk 1]
```diff
@@ -28,47 +30,52 @@ const styles = (theme: ThemeType): JssStyles => ({
...
-class EditTitle extends Component<EditTitleProps,{}> {
-  UNSAFE_componentWillMount() {
-    const { addToSuccessForm } = this.context
-    const { clearField } = this.props
-    addToSuccessForm(() => clearField())
-  }
-  render() {
-    const { document, value, path, placeholder, classes } = this.props
+const EditTitle = ({document, value, path, placeholder, updateCurrentValues, classes}: {
+    document: PostsBase,
```
Comment (jimrandomh) [inline packages/lesswrong/components/editor/EditTitle.tsx:34]: Fix indentation

File: packages/lesswrong/components/editor/EditTitle.tsx [hunk 2]
```diff
@@ -28,47 +30,52 @@ const styles = (theme: ThemeType): JssStyles => ({
...
+    value: any,
+    path: string,
+    placeholder: string,
+    updateCurrentValues: Function,
+    classes: ClassesType
+  }) => {
+    const { flash } = useMessages()
+  const [currentTitle, setCurrentTitle] = useState<string>(document.title)
```
Comment (jimrandomh) [inline packages/lesswrong/components/editor/EditTitle.tsx:42]: Rename to `lastSavedTitle`

Comment (darkruby501) [inline packages/lesswrong/components/editor/EditTitle.tsx:42]: change to lastSavedTitle

File: packages/lesswrong/components/notifications/CommentOnYourDraftNotificationHover.tsx [hunk 3]
```diff
@@ -30,7 +32,7 @@ const CommentOnYourDraftNotificationHover = ({notification, classes}: {
   return <div className={classes.root}>
     <div>
       {senderUserId ? <UsersName documentId={notification.extraData.senderUserID}/> : "Someone"}
-      {" commented on your draft "}
+      {(currentUser?._id !== post?.userId) ? " replied to your comment on " : " commented on your draft "}
```
Comment (jimrandomh) [inline packages/lesswrong/components/notifications/CommentOnYourDraftNotificationHover.tsx:35]:
```suggestion
      {(currentUser?._id !== post?.userId) ? " replied to your comment on " : ` commented on your draft ${postOrDraft}`}
```

Comment (darkruby501) [inline packages/lesswrong/components/notifications/CommentOnYourDraftNotificationHover.tsx:35]: Can check whether it's a draft or not.

File: packages/lesswrong/components/notifications/CommentOnYourDraftNotificationHover.tsx [hunk 4]
```diff
@@ -30,7 +32,7 @@ const CommentOnYourDraftNotificationHover = ({notification, classes}: {
   return <div className={classes.root}>
```
Comment (jimrandomh) [inline packages/lesswrong/components/notifications/CommentOnYourDraftNotificationHover.tsx:32]:
```suggestion
  const postOrDraft = post.draft ? "draft" : "post";
  
  return <div className={classes.root}>
```

**Review by jimrandomh — APPROVED** [review]

**Pull request merged**